react/features/polls/components/PollDialog.js

```diff
@@ -0,0 +1,363 @@
+
+        const initialState = this._clearState();
+
+        this.state = {
+            ...initialState
+        };
+    }
+
+    _closeDialog: () => void;
+
+    /**
+     * Callback invoked to close the dialog without saving changes.
+     *
+     * @private
+     * @returns {void}
+     */
+    _closeDialog() {
+        this.props.dispatch(hideDialog());
+    }
+
+    /**
+     * Component render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { isPollRunning, userID } = this.props;
+        const title = isPollRunning ? 'dialog.endPoll' : 'dialog.startPoll';
+        const tabs = isPollRunning ? [ {
+            component: VoteForm,
+            label: 'polls.vote',
+            props: {
+                userID,
+                onVoteClicked: this._vote
+            },
+            submit: null
+        } ] : [ {
+            component: PollCreateForm,
+            label: 'polls.create',
+            props: {
+                choices: this.state.choices,
+                onAddChoice: this._addChoice,
+                onChoiceTextChange: this._choiceTextChange,
+                onRemoveChoice: this._removeChoice,
+                onQuestionTextChange: this._onQuestionTextChange
+            },
+            propsUpdateFunction: (oldProps, newProps) => {
+                return { ...newProps };
+            },
+            submit: null
+        } ];
+
+        tabs.push({
+            component: PollResultsForm,
+            label: 'polls.results',
+            props: {},
+            submit: null
+        });
+
+        return (
+            <DialogWithTabs
+                closeDialog = { this._closeDialog }
+                cssClassName = { 'polls-dialog' }
+                key = { isPollRunning.toString() }
+                okTitleKey = { title }
+                onSubmit = { this._onSubmit }
+                tabs = { tabs }
+                titleKey = { 'dialog.polls' } />
+        );
+    }
+
+    _onSubmit: () => void;
+
+    /**
+     * Submit button handler.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        const { isPollRunning } = this.props;
+
+        if (isPollRunning) {
+            this._endPoll();
+        } else {
+            this._createNewPoll();
+        }
+    }
+
+    _addChoice: (void) => void;
+
+    /**
+     * Add new item to the list.
+     *
+     * @returns {void}
+     */
+    _addChoice() {
+        const id = v1();
+
+        this.setState({
+            choices: {
+                ...this.state.choices,
+                [id]: {
+                    id,
+                    text: '',
+                    votes: []
+                }
+            }
+        });
+    }
+
+    _removeChoice: (number) => void;
+
+    /**
+     * Removes an item from list.
+     *
+     * @param {string} id - ID of item in list.
+     * @returns {void}
+     */
+    _removeChoice(id: string) {
+        const filteredChoices = Object.assign({}, this.state.choices);
```

Comment:
Generally the style { ...this.state.choices } would be used, like you did in the constructor.